addon/components/bs-link-to.js

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable ember/classic-decorator-no-classic-methods */
+import Component from '@ember/component';
+import { tagName } from '@ember-decorators/component';
+import { computed } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { assert } from '@ember/debug';
+
+/**
+ This is largely copied from Ember.LinkComponent. It is used as extending from Ember.LinkComponent has been deprecated.
+ We need this to
+ * register ourselves to a parent component that needs to know `active` state due to Bootstrap markup requirements, see Nav/LinkTo
+ * continue supporting positional params until we can remove support
+
+ @class LinkComponent
+ @namespace Components
+ @extends Component
+ @private
+*/
+@tagName('')
+class LinkComponent extends Component {
+  // We still need to use the private service, so this component can react to changes of `currentState`, e.g. changing
+  // the model while the route in unchanged.
+  // eslint-disable-next-line ember/no-private-routing-service
+  @service('-routing')
+  _routing;
+
+  @computed('models', 'query', 'route', '_routing.currentState', 'current-when')
+  get active() {
+    let state = this._routing.currentState;
+
+    if (state) {
+      return this._isActive(state);
+    } else {
+      return false;
+    }
+  }
+
+  @computed('model', 'models')
+  get _models() {
+    let { model, models } = this;
+
+    if (model !== undefined) {
+      return [model];
+    } else if (models !== undefined) {
+      assert('The `@models` argument must be an array.', Array.isArray(models));
+      return models;
+    } else {
+      return [];
+    }
+  }
+
+  _isActive(routerState) {
+    let currentWhen = this['current-when'];
+
+    if (typeof currentWhen === 'boolean') {
+      return currentWhen;
+    }
+
+    let { _models: models, _routing: routing } = this;
+
+    if (typeof currentWhen === 'string') {
+      return currentWhen
+        .split(' ')
+        .some((route) => routing.isActiveForRoute(models, undefined, this._namespaceRoute(route), routerState));
+    } else {
+      return routing.isActiveForRoute(models, this.query || {}, this.route, routerState);
+    }
+  }
+
+  // eslint-disable-next-line ember/no-component-lifecycle-hooks
+  didReceiveAttrs() {
+    super.didReceiveAttrs(...arguments);
+
+    let { params } = this;
+    if (!params || params.length === 0) {
+      return;
+    }
+
+    params = params.slice();
```

Comment:
We haven't  formally deprecated positional params for link params it seems. While this is now deprecated for Ember's original LinkComponent, we do not trigger this deprecation anymore. So we should deprecate this explicitly by ourselves I think. Should be a follow up PR.

Reply:
> Haven't we dropped support for curly-bracket component invocation in v4

We haven't declared so, no. We dropped arguments for HTML attributes, and more or less recommended to use angle brackets, but haven't explicitly stopped support for curly invocation. At the time there was no reason for this, and this is still not deprecated from Ember's side AFAIK. Except for this positional arguments issue there is no reason why curly invocation should bother us, as long as Ember supports it we can basically not care about it. 

We can choose to drop support in a next major, or just say that the use of our custom link component drops support for positional arguments and all the other things (currentWhen) that accidentally became part of our implementation due to extending from LinkComponent, whether or not we consider this a public API or not. (it's a "gray area" I guess)

IIRC you were more on the side of accepting major versions more often, right? So should we actually do this refactoring as part of a v5, with just this edge case-y thing as the (potentially, but rather rare) breaking change?